lib/Controller/SocialApiController.php

```diff
@@ -0,0 +1,288 @@
+<?php
+/**
+ * @copyright Copyright (c) 2020 Matthias Heinisch <nextcloud@matthiasheinisch.de>
+ *
+ * @author Matthias Heinisch <nextcloud@matthiasheinisch.de>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Contacts\Controller;
+
+use OCP\AppFramework\ApiController;
+use OCP\AppFramework\Http;
+use OCP\AppFramework\Http\DataResponse;
+use OCP\AppFramework\Http\JSONResponse;
+use OCP\AppFramework\Http\TemplateResponse;
+// use OCP\IInitialStateService;
+use OCP\IConfig;
+use OCP\Contacts\IManager;
+use OCP\L10N\IFactory;
+use OCP\IRequest;
+
+
+class SocialApiController extends ApiController {
+
+	protected $appName;
+
+	//** @var IInitialStateService */
+	// private  $initialStateService;
+
+	/** @var IFactory */
+	private  $languageFactory;
+	/** @var IManager */
+	private  $manager;
+	/** @var IConfig */
+	private  $config;
+
+	/**
+	 * This constant stores the supported social networks
+	 * It is an ordered list, so that first listed items will be checked first
+	 * Each item stores the avatar-url-formula as recipe, a cleanup parameter to
+	 * extract the profile-id from the users entry, and possible filters to check
+	 * validity
+	 * 
+	 * @const {array} SOCIAL_CONNECTORS dictionary of supported social networks
+	 */
+	const SOCIAL_CONNECTORS = [
+		'facebook' 	=> [
+			'recipe' 	=> 'https://graph.facebook.com/{socialId}/picture?width=720',
+			'cleanups' 	=> ['basename'],
+			'checks'	=> [],
+		],
+		'tumblr' 	=> [
+			'recipe' 	=> 'https://api.tumblr.com/v2/blog/{socialId}/avatar/512',
+			'cleanups' 	=> ['basename'],
+			'checks'	=> [],
+		],
+		/* do we trust avatars.io?
+		'instagram' 	=> [
+			'recipe' 	=> 'http://avatars.io/instagram/{socialId}',
+			'cleanups' 	=> ['basename'],
+			'checks'	=> []
+		],
+		'twitter' 	=> [
+			'recipe' 	=> 'http://avatars.io/twitter/{socialId}',
```

Comment:
If you want to dive into this now, you can use this: 
https://developer.twitter.com/en/docs/accounts-and-users/user-profile-images-and-banners

But i'm fine having this in without twitter first :)

Reply:
Yes - but as I understand it, it always requires authentication.

Reply:
Yep :)
There are ways around it, but not ideal.
Let's see later!